src/rgw/rgw_log.cc

```diff
@@ -226,6 +226,18 @@ static void log_usage(struct req_state *s, const string& op_name)
   utime_t ts = ceph_clock_now();
 
   usage_logger->insert(ts, entry);
+
+  if(s->cct->_conf->rgw_enable_usage_log_at_subuser_level) {
+    const auto subuser = s->auth.identity->get_subuser_name();
+    string subuser_name = *subuser;
+
+    if (!subuser_name.empty()) {
+      ldout(s->cct, 5) << "usage log subuser entry: " << subuser_name << dendl;
+      rgw_usage_log_entry sentry(u, p, bucket_name, subuser_name);
+      sentry.add(op_name, data);
+      usage_logger->insert(ts, sentry);
```

Comment:
Looks conceptually OK if the potential inconsistencies between stats of a given `user` and its `subusers` aren't an issue. Most likely there is no atomicity guarantee across the two `insert` calls.